### PR TITLE
refactor: use nvim_create_autocmd api replace vim.cmd

### PR DESCRIPTION
**Pull request by glepnir**
use `nvim_create_autocmd` .

### CODE REVIEW

File: runtime/lua/vim/diagnostic.lua [hunk 1]
```diff
@@ -347,17 +347,13 @@ local function schedule_display(namespace, bufnr, args)
 
   local key = make_augroup_key(namespace, bufnr)
   if not registered_autocmds[key] then
-    vim.cmd(string.format(
-      [[augroup %s
-      au!
```
Comment (clason) [inline runtime/lua/vim/diagnostic.lua:352]: your refactor is missing the `au!`, i.e., clearing previous autocommands in the augroup. You need to clear them manually.

File: runtime/lua/vim/lsp/util.lua [hunk 2]
```diff
@@ -1366,29 +1366,21 @@ local function close_preview_autocmd(events, winnr, bufnrs)
 
   -- close the preview window when entered a buffer that is not
   -- the floating window buffer or the buffer that spawned it
-  vim.cmd(string.format(
-    [[
-    augroup %s
-      autocmd!
```
Comment (clason) [inline runtime/lua/vim/lsp/util.lua:1372]: same

File: runtime/lua/vim/lsp/util.lua [hunk 3]
```diff
@@ -1366,29 +1366,21 @@ local function close_preview_autocmd(events, winnr, bufnrs)
...
-      autocmd BufEnter * lua vim.lsp.util._close_preview_window(%d, {%s})
-    augroup end
-  ]],
-    augroup,
-    winnr,
-    table.concat(bufnrs, ',')
-  ))
+  api.nvim_create_autocmds('BufEnter', {
+    group = augroup,
+    pattern = '*',
```
Comment (clason) [inline runtime/lua/vim/lsp/util.lua:1371]: `pattern='*'` is the default, no need to add that.

File: runtime/lua/vim/diagnostic.lua (hunk 1 above)
Comment (glepnir) [reply runtime/lua/vim/diagnostic.lua:352]: use `nvim_del_autocmd`  to clear prev autocmd?

File: runtime/lua/vim/lsp/util.lua (hunk 3 above)
Comment (glepnir) [reply runtime/lua/vim/lsp/util.lua:1371]: got it

File: runtime/lua/vim/diagnostic.lua (hunk 1 above)
Comment (max397574) [reply runtime/lua/vim/diagnostic.lua:352]: can't you just create an augroup with `clear = true` (which should be default iirc)

Comment (clason) [reply runtime/lua/vim/diagnostic.lua:352]: use `nvim_clear_autocmds()` (see the similar refactor at https://github.com/nvim-treesitter/nvim-treesitter/commit/07eb437bb1c4c0a0298553dd509294f82362d6bc)

Comment (glepnir) [reply runtime/lua/vim/diagnostic.lua:352]: yup check doc I found `clear = true`

**Pull request closed without merging**